slsim/lens_pop.py

```diff
@@ -99,9 +111,20 @@ def __init__(
             )
             self._source_model_type = "extended"
         elif source_type == "quasars":
-            from sim_pipeline.Sources.quasars import Quasars
-
-            self._sources = Quasars(cosmo=cosmo, sky_area=sky_area)
+            from slsim.Sources.quasars import Quasars
+            from slsim.Sources.quasar_catalog.simple_quasar import quasar_catalog
+
+            if kwargs_quasars is None:
+                kwargs_quasars = {}
+            quasar_source = quasar_catalog(**kwargs_quasars)
+            column_names = quasar_source.colnames
+            if (
+                "mag_r" not in column_names
+                and "mag_i" not in column_names
+                and "mag_g" not in column_names
+            ):
+                raise ValueError("required parameters missing in quasar_list columns")
```

Comment:
what if we only want one specific band (which perhaps can even be another band? This should only raise errors when calling functions that request these properties

Reply:
Transfered these to source function where we call for source magnitude.